Remove/fix all references to undefined Kconfig symbols in Kconfig* files

ulfalizer opened the pull request:
This series removes/fixes all references to undefined symbols within `Kconfig*` files in Zephyr. The plan is to turn undefined references into an error later.

Additional cleanup might be possible here.

When fixing the two malformed `default n` properties, I removed them instead. Kconfig `bool` symbols implicitly default to `n`, which might be poorly documented. Please complain if you prefer to keep the `default`s, e.g. for consistency with surrounding code.

@MaureenHelm: could you check commit 88ad0dc57028 ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those `default`s were on can be removed?

ulfalizer committed:
arch: arc: Fix reference to removed NSIM Kconfig symbol

The NSIM symbol was removed in commit 9bc69a46fa09 ("boards: Update arc
em_starterkit support from 2.2 to 2.3").

Guess that UART_NSIM should be used here now, similar to what was done
in commit 45221a97064d ("serial: nsim: Fix impossible-to-enable
CONFIG_UART_NSIM").

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
drivers: sensors: Consistently quote "GPIO_0" string default

These cases weren't caught by the heuristic used by the warning added in
commit 6eabea3a7e6f ("Kconfiglib: Warn for unquoted string defaults"),
because "GPIO_0" has no lowercase characters in it.

Unquoted Kconfig values are are indistinguishable from reference to
(undefined) symbols in general. Quoting all string defaults will help
find "true" references to undefined Kconfig symbols, and makes it
clearer that the value is constant.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
lib: Fix malformed JSON_LIBARY Kconfig default

'default N' should have been 'default n', though they happen to have the
same effect here, due to undefined Kconfig symbols ('N') evaluating to
'n' in a boolean sense.

Kconfig bool symbols implicitly default to 'n', so remove the default
rather than fixing it.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
net: rpl: Fix malformed Kconfig default

'default false' should have been 'default n', though they happen to have
the same effect here, due to undefined Kconfig symbols ('false')
evaluating to 'n' in a boolean sense.

Kconfig bool symbols implicitly default to 'n', so remove the default
rather than fixing it.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
ext: lib: mgmt: Remove MDLOG Kconfig reference

This symbol never seems to have been defined.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
kconfig: Remove UART_QMSI_{0,1}_NAME Kconfig reference

This symbol was removed by commit 17c6456678d5 ("drivers/uart: Use dts
to set uart priorities for QMSI driver"). The setting should come from
DTS now.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
drivers: Replace ff hex constants with 0xff

This makes it easier to distinguish them from "true" undefined symbols.

Internally, all int/hex literals are treated as undefined symbols, which
always get their name as their value. The C tools work the same way.

The plan is to turn references to undefined Kconfig symbols into an
error later.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
CC @nashif @carlescufi @MaureenHelm @jhedberg 

No hurry if you're busy with the release.

MaureenHelm commented on subsys/net/ip/Kconfig.rpl:
```diff
@@ -232,7 +232,6 @@ config	NET_RPL_PREFERENCE
 config	NET_RPL_GROUNDED
 	bool "DAG grounded default value"
 	depends on NET_RPL
-	default false
```
It looks this change goes with the previous commit's message, and vice versa?

ulfalizer replied:
Oops... git slip-up. Fixed.

MaureenHelm reviewed (commented):
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?

LGTM

ulfalizer commented:
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?
> 
> LGTM

Removed now

ulfalizer commented on ext/lib/mgmt/mcumgr/cmd/log_mgmt/Kconfig:
```diff
@@ -18,7 +18,6 @@
 menuconfig MCUMGR_CMD_LOG_MGMT
     bool
     prompt "Enable mcumgr handlers for log management"
-    depends on MDLOG
```
As a note here, [MCUMGR_CMD_LOG_MGMT](http://docs.zephyrproject.org/reference/kconfig/CONFIG_MCUMGR_CMD_LOG_MGMT.html) (*Enables mcumgr handlers for log management*) will have been impossible to enable until this gets in, from the moment it got merged.

ulfalizer commented:
CC @galak

MaureenHelm commented:
> @MaureenHelm: could you check commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") and see if the symbols that those defaults were on can be removed?
> LGTM
> 
> Removed now

@ulfalizer I misunderstood your question before. The defaults can be removed, but I think @tbursztyka or @jfischer-phytec-iot may still need the symbols they were on. In other words, your original commit 88ad0dc ("drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references") looked good, but I'm not sure about commit 7112c670df1727acc69bb5ed01700c0e8e6603b7

ulfalizer committed:
drivers: ieee802154: Remove GPIO_MCUX_PORT{A,B}_NAME Kconfig references

These symbols were removed by commit 4e8f29f31954 ("gpio: Refactor the
mcux gpio driver to use dts"). The settings should come from DTS now.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
drivers: gpio: Fix GPIO_QMSI_{0,1}_NAME Kconfig references

These symbols were removed by commit ed26b95 ("drivers/gpio:
Removing dts generated options in QMSI Kconfig"). The settings should
come from DTS now.

Commit 9611002f7a79f ("boards/x86: Use right GPIO names for CC2520")
fixed some of the references, but there are still others left. Guess
that the same fix is appropriate in those spots (hardcoded
"GPIO_0"/"GPIO_1" names).

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
@MaureenHelm 
Changed it to just remove the defaults. See if it looks alright now.

ulfalizer commented:
@tbursztyka 
Could you check if https://github.com/zephyrproject-rtos/zephyr/pull/8106/commits/516e602bcbbc8fd5445ac80ce65ca8e271fdb3df looks okay?

ulfalizer commented:
@tbursztyka
Could you check if https://github.com/zephyrproject-rtos/zephyr/pull/8106/commits/5fff025b7b7010124a7ca62e43cf08c135b41dea looks okay?

ulfalizer commented:
*Ping*

carlescufi reviewed (approved):
looks good, thanks for the cleanup!
